python/pyspark/ml/tests/test_stat.py

```diff
@@ -40,7 +40,7 @@ def test_chisquaretest(self):
 
 
 if __name__ == "__main__":
-    from pyspark.ml.tests.test_stat import *
+    from pyspark.ml.tests.test_stat import *  # noqa: F401
```

Comment:
@srowen We could consider removing these ones. There are some entries for running the tests using the xmlrunner. However it is unclear to me who's using this, maybe the CI?

Reply:
I think it's not controversial to remove truly unused imports. If you have a PR that just does that, I'd support it. adding the pep rule and exclusions may just not be worth it now.

Reply:
Can you move this forward to test? I can check if I remove the xmlrunner, what the impact is on the tests by taking a diff of the Jenkins logs.

Reply:
I can trigger tests though perhaps better to not try enforcing/excluding the import rule now.

Reply:
Thanks! I've send a mail on the devlist: http://apache-spark-developers-list.1001551.n3.nabble.com/Python-xmlrunner-being-used-td29896.html

Reply:
No, no. don't remove this. It will keep the source code path in console output. It is also used in coverage report as well. XML reporter is used in Jenkins.

Reply:
Thanks @HyukjinKwon. It looks like it is being used by Jenkins indeed. I will try to remove the imports to see if the tests are still being picked up by Jenkins. These imports look redundant since it is importing the file itself 🤔

Reply:
Ah, these imports actually makes the output file paths correct. For example, if you run the unittests, the console outputs are like:

```
test_blahblah (pyspark.sql.blahblah.BlahBlahTests) ... ok

----------------------------------------------------------------------
Ran 3 tests in 0.001s

OK
```

if you remove the imports, the output becomes:

```
test_blahblah (__main__.BlahBlahTests) ... ok

----------------------------------------------------------------------
Ran 3 tests in 0.001s

OK
```

It's related to who we run the tests via our custom script `python/run-tests.py`. Let's leave them as are for now.

Reply:
Ack! Missed that one. I've reverted the imports :)